src/util/client_proxy.rs

```diff
@@ -81,6 +80,12 @@ impl<R, S, E: From<io::Error>> Service for ClientProxy<R, S, E> {
     }
 }
 
+impl<R, S, E> fmt::Debug for ClientProxy<R, S, E> {
```

Comment:
I'd recommend eagerly bounding types by `fmt::Debug` even if they're not used yet, that way we can expand one day to using them if need be.